include/zephyr/net/ethernet.h

```diff
@@ -467,6 +469,24 @@ struct ethernet_txtime_param {
 
 /** @cond INTERNAL_HIDDEN */
 
+/** Protocols that are supported by checksum offloading */
+enum ethernet_checksum_support {
+	ETHERNET_CHECKSUM_SUPPORT_NONE		  = 0,
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER = BIT( 0),
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_TCP    = BIT( 1),
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_UDP    = BIT( 2),
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_ICMP   = BIT( 3),
+	/** Space for future protocols and restrictions for IPV4 */
+	ETHERNET_CHECKSUM_SUPPORT_IPV6_HEADER = BIT(16),
+	ETHERNET_CHECKSUM_SUPPORT_IPV6_TCP    = BIT(17),
+	ETHERNET_CHECKSUM_SUPPORT_IPV6_UDP    = BIT(18),
```

Comment:
Do we need to differentiate IPv4/6 TCP/UDP? I'd stick to the originally proposed list: IPv4/6, TCP, UDP, ICMPv4/6.

Reply:
OK, I will merge the UDP and TCP for IPv4/6.